authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/RangerHelper.java

```diff
@@ -0,0 +1,533 @@
+                          RangerPolicy.RangerPolicyItem policyItem =
+                              new RangerPolicy.RangerPolicyItem();
+                          RangerPolicy.RangerPolicyItemAccess access =
+                              new RangerPolicy.RangerPolicyItemAccess();
+                          access.setType(mappedPrivilege);
+                          policyItem.getAccesses().add(access);
+                          policyItem.getRoles().add(roleName);
+                          if (Privilege.Condition.ALLOW == gravitinoPrivilege.condition()) {
+                            policy.getPolicyItems().add(policyItem);
+                          } else {
+                            policy.getDenyPolicyItems().add(policyItem);
+                          }
+                        } else {
+                          // If the policy item exists, then add the role to the policy item
+                          matchPolicyItems.stream()
+                              .forEach(
+                                  policyItem -> {
+                                    // If the role is not in the policy item, then add it
+                                    if (!policyItem.getRoles().contains(roleName)) {
+                                      policyItem.getRoles().add(roleName);
+                                    }
+                                  });
+                        }
+                      });
+            });
+  }
+
+  /**
+   * Remove policy item base the securable object's privileges and role name. <br>
+   * We cannot directly clean the policy items because one Ranger policy maybe contains multiple
+   * Gravitino privilege objects. <br>
+   */
+  void removePolicyItem(RangerPolicy policy, String roleName, SecurableObject securableObject) {
+    // First check the privilege if support in the Ranger Hive
+    checkPrivileges(securableObject);
+
+    // Delete the policy role base the securable object's privileges
+    policy.getPolicyItems().stream()
+        .forEach(
+            policyItem -> {
+              policyItem
+                  .getAccesses()
+                  .forEach(
+                      access -> {
+                        boolean matchPrivilege =
+                            securableObject.privileges().stream()
+                                .filter(Objects::nonNull)
+                                .flatMap(
+                                    privilege ->
+                                        rangerAuthorizationPlugin
+                                            .translatePrivilege(privilege.name()).stream())
+                                .filter(Objects::nonNull)
+                                .anyMatch(
+                                    privilege -> {
+                                      return access.getType().equals(privilege);
+                                    });
+                        if (matchPrivilege
+                            && !policyItem.getUsers().isEmpty()
```

Comment:
Why do we need `getUsers` is empty and `getGroups` is empty?

Reply:
We just remove roles here, not remove policyItem.

Reply:
Yes, you are right.
I fix this problem and add a test case.